content/docs/authentication/access_tokens_guard.md

```diff
@@ -539,6 +539,20 @@ import router from '@adonisjs/core/services/router'
 const SessionController = () => import('#controllers/session_controller')
 
 router.post('session', [SessionController, 'store'])
-router.destroy('session', [SessionController, 'destroy'])
+router.delete('session', [SessionController, 'delete'])
   .use(middleware.auth({ guards: ['api'] }))
 ```
+
+:::warning
+
+Use [content negotiation](../authentication/verifying_user_credentials.md#handling-exceptions) to get appropriate responses when `User.verifyCredentials` fails (and throws [E_INVALID_CREDENTIALS](../references/exceptions#e_invalid_credentials)).
+
+In the above example's case, the client should include an `Accept=application/json` header in post requests to `/session`. This ensures that failures will result in json formatted responses rather than redirects.
+
+:::
+
+:::warning
+
+If [CSRF](../security/securing_ssr_applications.md#csrf-protection) protection is enabled, remember to add an exception for the `/session` route. See [shield config reference](https://docs.adonisjs.com/guides/security/securing-ssr-applications#config-reference)
```

Comment:
Is there any reason the `/session` route should be exempted from the CSRF validation, given if all other post submissions use CSRF?

Reply:
Good point. If you are using a regular browser with cookies enabled, there should be no reason to disable CSRF protection.

I didn't think about that scenario since I mostly see the usage of access tokens for regular sessions as a fallback for situations where you do not have cookies available. For instance if signing in from an app.

Let me change the content so that it's more clear that CSRF should only be disabled for such sessions.

Reply:
But if you are using an app then I think all the form submissions from the mobile app should disable CSRF. In that case, this warning/note has to be in a broader context with all forms and not just the login form.

Reply:
> But if you are using an app then I think all the form submissions from the mobile app should disable CSRF. In that case, this warning/note has to be in a broader context with all forms and not just the login form.

I was thinking the same. I have updated the section (which I have changed to a tip), to try and communicate that in a concise way.